### ISSUE TITLE
0.7.1 resolver breaks 1.13.x app

### ISSUE THREAD

**Issue by blimmer** [minutebase/ember-can#32]
Getting this issue with 0.7.1 on 1.13.x app:

```
not ok 60 PhantomJS 2.0 - TestLoader Failures cs/tests/acceptance/error-test: could not be loaded
    ---
        message: >
            Could not find module `ember-resolver` imported from `cs/initializers/setup-ember-can`
        stack: >
            	at missingModule (http://localhost:7357/assets/vendor.js:177:93)
            	at findModule (http://localhost:7357/assets/vendor.js:192:30)
            	at reify (http://localhost:7357/assets/vendor.js:128:32)
            	at build (http://localhost:7357/assets/vendor.js:146:28)
            	at findModule (http://localhost:7357/assets/vendor.js:194:14)
            	at requireModule (http://localhost:7357/assets/vendor.js:181:22)
            	at http://localhost:7357/assets/vendor.js:70968:33
            	at forEach ([native code])
            	at default (http://localhost:7357/assets/vendor.js:70965:19)
            	at http://localhost:7357/assets/cs.js:132:41
            	at exports (http://localhost:7357/assets/vendor.js:96:39)
            	at build (http://localhost:7357/assets/vendor.js:146:17)
            	at findModule (http://localhost:7357/assets/vendor.js:194:14)
            	at reify (http://localhost:7357/assets/vendor.js:128:32)
            	at build (http://localhost:7357/assets/vendor.js:146:28)
            	at findModule (http://localhost:7357/assets/vendor.js:194:14)
            	at reify (http://localhost:7357/assets/vendor.js:128:32)
            	at build (http://localhost:7357/assets/vendor.js:146:28)
            	at findModule (http://localhost:7357/assets/vendor.js:194:14)
            	at requireModule (http://localhost:7357/assets/vendor.js:181:22)
            	at require (http://localhost:7357/assets/test-loader.js:60:16)
            	at loadModules (http://localhost:7357/assets/test-loader.js:51:25)
            	at load (http://localhost:7357/assets/test-loader.js:82:35)
            	at http://localhost:7357/assets/test-support.js:15143:20
        Log: |
```

**Comment by rlivsey** [comment]
Ah crap pulled the trigger a bit soon on that.

Hmm is there a nice way of detecting this so we can avoid the deprecation on new apps but not make older apps unhappy? Or maybe call this a backwards incompatible change and go to 0.8.0 (releasing 0.7.2 with this reverted)?

**Comment by blimmer** [comment]
Sorry, I'm a bit out of the loop on that one - is the new ember resolver thing a 2.x change? I don't have any apps on 2.x yet

**Comment by blimmer** [comment]
It'd also be a bit of a bummer to make this entire addon backwards-incompatible with Ember 1.x to resolve a deprecation warning in 2.x.

**Comment by rlivsey** [comment]
It looks like it should work fine in 1.x too with a few changes.

@rwjblue links to a diff of the changes required here: https://github.com/ember-cli/ember-cli/pull/4777#issuecomment-136423099

Would be nice to issue a warning maybe, or fallback to the old `ember/resolver` but IIRC you can't conditionally require in ES6.

**Comment by rwjblue** [comment]
You can drop down and do non-import stuff via require. 

    requirejs.entries['ember-resolver']

^^ can be used to see if that module is present, and:


    var Resolver = require(moduleName)['default']

Will give you the resolver from the module once you figure out which one it is in.

**Comment by rlivsey** [comment]
@rwjblue aha awesome, thanks!

It's late here and I'm heading to bed now, but should be able to find some time tomorrow to take a look at this. @blimmer feel free to take a look if you get a chance in the meantime and I'll get a 0.7.2 out asap.

**Comment by blimmer** [comment]
Sounds good. Might want to unpublish 0.7.1 so we don't have to rush on 0.7.2.

**Comment by rlivsey** [comment]
@blimmer fair play, 0.7.1 is no more :boom:

**Comment by rwjblue** [comment]
Submitted https://github.com/minutebase/ember-can/pull/33

**Comment by rlivsey** [comment]
Released 0.7.2, thanks @rwjblue & @blimmer
